staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,414 @@
+}
+
+// PriorityLevelConfigurationReference contains information that points to the "request-priority" being used.
+type PriorityLevelConfigurationReference struct {
+	// `name` is the name of resource being referenced
+	// Required.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PolicyRulesWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
+// making the request, the verb being requested, and the resource to be acted upon. This PolicyRulesWithSubjects matches
+// a request if and only if both (a) at least one member of subjects matches the request and (b) at least one member
+// of resourceRules or nonResourceRules matches the request.
+type PolicyRulesWithSubjects struct {
+	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
+	// There must be at least one subject for this field, or it will be rejected by the api validation.
+	// Required.
+	Subjects []Subject `json:"subjects" protobuf:"bytes,1,rep,name=subjects"`
+	// `resourceRules` is a slice of ResourcePolicyRules that identify matching requests according to their verb and the
+	// resource they request to act upon.
+	// +optional
+	ResourceRules []ResourcePolicyRule `json:"resourceRules,omitempty" protobuf:"bytes,2,opt,name=resourceRules"`
+	// `nonResourceRules` is a list of NonResourcePolicyRules that identify matching requests according to their verb
+	// and the non-resource URL they request to act upon.
+	// +optional
+	NonResourceRules []NonResourcePolicyRule `json:"nonResourceRules,omitempty" protobuf:"bytes,3,opt,name=nonResourceRules"`
+}
+
+// Subject matches a set of users.
+// Syntactically, Subject is a general API object reference.
+// Authorization produces a username and a set of groups, and we imagine special kinds of non-namespaced objects,
+// User and Group to represent such a username or group.
+// The only kind of true object reference that currently will match any users is ServiceAccount.
+// +union
+type Subject struct {
+	// Required
+	// +unionDiscriminator
+	Kind SubjectKind `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+	// +optional
+	User *UserSubject `json:"user,omitempty" protobuf:"bytes,2,opt,name=user"`
+	// +optional
+	Group *GroupSubject `json:"group,omitempty" protobuf:"bytes,3,opt,name=group"`
+	// +optional
+	ServiceAccount *ServiceAccountSubject `json:"serviceAccount,omitempty" protobuf:"bytes,4,opt,name=serviceAccount"`
+}
+
+type SubjectKind string
+
+// Supported subject's kinds.
+const (
+	SubjectKindUser           SubjectKind = "User"
+	SubjectKindGroup          SubjectKind = "Group"
+	SubjectKindServiceAccount SubjectKind = "ServiceAccount"
+)
+
+// UserSubject holds detailed information for user-kind subject.
+type UserSubject struct {
+	// Required
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// GroupSubject holds detailed information for group-kind subject.
+type GroupSubject struct {
+	// Required
```

Comment:
This comment to start in the usual way, with the name of the field being documented.  And we should explain the wildcard.  Thus I suggest something like the following.

> // `name` is the user group that matches.  `"*"` matches all user groups.
> // Required.

Reply:
applied suggestion